src/Countries/Ecuador.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace Spatie\Holidays\Countries;
+
+use Carbon\CarbonImmutable;
+
+class Ecuador extends Country
+{
+    public function countryCode(): string
+    {
+        return 'ec';
+    }
+
+    protected function allHolidays(int $year): array
+    {
+        return array_merge([
+            'Año Nuevo' => '01-01',
+        ], $this->variableHolidays($year));
+    }
+
+    public function nearestDay(int $year, int $month, int $day)
+    {
+        $date = CarbonImmutable::createFromDate($year, $month, $day);
+
+        if($date->is('Tuesday') || $date->is('Saturday')) {
+            return $date->subDay();
+        }
+
+        if($date->is('Sunday')) {
+            return $date->addDay();
+        }
+
+        if($date->is('Wednesday') || $date->is('Thursday')) {
+            return $date->next(CarbonImmutable::FRIDAY);
+        }
+
+        return $date;
+    }
+
+    public function getChristmasHoliday(int $year)
+    {
+        $date = CarbonImmutable::createFromDate($year, 12, 25);
+
+        if($year === 2022) {
```

Comment:
Do you need this behaviour?
https://github.com/spatie/holidays/blob/main/src/Concerns/Observable.php#L36

Reply:
Yes, I already implemented it